Updates the activation flow for Jetpack manage

enejb opened the pull request:
- Display a nicer message with a link to go back.
- Customize the message for each of the different cases when for when
the user activates lands on this page.

**Todo**
- [x] Update the language for each of the cases.
- [x] Add icons as well for each of the cases.
- [x] Move the styles to its the admin.scss file. 

**To test**
Disable Manage and visit 
wp-admin/admin.php?page=jetpack&configure=manage 
you can test the wording of each of the different cases by adding 
&section=plugins or &section=menus or &section=themes or &section=themes or &section=security-settings or just leave that part off. 

When you have Manage enabled you will be able to visit
The same page and the same sections but now the confirmation page will be there.

jeherve commented on class.jetpack.php:
```diff
@@ -4494,7 +4505,12 @@ public static function admin_screen_configure_module( $module_id ) {
 		if ( ! in_array( $module_id, Jetpack::get_active_modules() ) && current_user_can( 'manage_options' ) ) {
-			self::display_activate_module_link( $module_id );
+			if ( has_action( 'display_activate_module_setting_'.$module_id ) ) {
+				do_action( 'display_activate_module_setting_'.$module_id );
```
Could you add a docblock here? THanks!

jeherve commented on modules/manage/activate-admin.php:
```diff
@@ -0,0 +1,57 @@
+$section = isset( $_GET['section'] ) ? $_GET['section'] : null;
+
+switch( $section ) {
+	case 'plugins':
```
The wording seems a bit weird for these notices. It might be worth reviewing them all, as it seems like we're begging folks to enable the module right now.

enejb replied:
I have updated the wording here.

jeherve commented on modules/manage/activate-admin.php:
```diff
@@ -0,0 +1,57 @@
+	<p style="max-width: 600px; text-align:center; font-size: 22px; color:#999; margin: 20px auto;">
+		<?php echo esc_html( $description ); ?>
+	</p>
+	<p style="text-align: center;">	<?php printf( '<a class="button-primary" style="font-size:16px; padding: 0 20px; height:40px; line-height:40px;"  href="%s">%s</a>',
```
I'd suggest using `%1$s`and `%2$s` here, to avoid any confusion if more arguments were to be added in the future.

jeherve commented on modules/manage/confirm-admin.php:
```diff
@@ -0,0 +1,70 @@
+	<div class="" style="width:150px; height: 150px; border-radius: 75px; margin: 20px auto; background:#A9C4D9;">
+		<?php echo $icon; ?>
+	</div>
+	<h1><span style="font-size: 38px; color: #81A944;" class="genericon genericon-checkmark"></span><?php echo esc_html( __( 'Enable Manage Enabled!', 'jetpack' ) ); ?></h1>
```
How about using `esc_html_e` here? That string seems a bit weird though, I think there is one extra "Enable".

alternatekev commented:
Text changes to make:

Plugins:
Header: Enable Jetpack Manage
Text: Just one more step before your plugins can be managed with WordPress.com. Click the button below and in no time, you'll be managing all of your plugins in one place.
Button: Enable Jetpack Manage Now

Menus:
Header: Enable Jetpack Manage
Text: Just one more step before your menus can be managed with WordPress.com. Click the button below and in no time, you'll be managing your menus with our newly-redesigned user interface.
Button: Enable Jetpack Manage Now

Themes:
Header: Enable Jetpack Manage
Text: Just one more step before your themes can be managed with WordPress.com. Click the button below and in no time, you'll be managing your themes with our newly-redesigned user interface.
Button: Enable Jetpack Manage Now

Security Settings:
Header: Enable Jetpack Manage
Text: Just one more step before your site can be secured by Jetpack and WordPress.com. Click the button below and in no time, you'll be safe and secure.
Button: Enable Jetpack Manage Now

alternatekev commented:
Follow-up (after Manage is enabled) page text changes:

Plugins/Menus/Themes/Security:
Header: Jetpack Manage Enabled
Text: Well that was pretty easy. Click the button below to go back to WordPress.com to manage all of your sites in one convenient place.

Button Text:
Plugins: Manage This Site's Plugins on WordPress.com
Menus: Manage This Site's Menus on WordPress.com
Themes: Manage This Site's Themes on WordPress.com
Security: Manage This Site's Security Features on WordPress.com

We may consider using the actual blog's name instead of "this site"?

justinkropp commented:
I'm on board with @alternatekev's changes, with just a couple thoughts/changes:

Should we avoid using the phrase _and in no time_? I only ask this because we might want to avoid stating absolutes here; it sounds a bit like a promise. I'd rather let the user discover that it's fast and easy than us promising it's fast and easy and we don't quite live up to their expectations of "in no time".

I'd like to change _our newly-redesigned user interface_ to read simply: "our redesign interface."

As an alternative to the messaging after _Manage_ is enabled:

**Header**: Jetpack Manage Enabled
**Text**: Well that was easy. You can now manage all of your sites in one convenient place on WordPress.com
**Button Text**: Manage Your Plugins, Manage Your Menus, etc.

@alternatekev Thoughts?

enejb commented:
I liked the simpler version that @justinkropp suggested. 

This is what the PR looks like right now. 

cc @alternatekev and @alternatekev. 
I think some of the descriptions are a bit long. 

I think adding icons would help here for sure :) 

Plugins:
![screen shot 2015-10-21 at 12 12 47 pm](https://cloud.githubusercontent.com/assets/115071/10649282/f5001d14-77f7-11e5-9fdd-c1e2bb675245.png)
![plugins](https://cloud.githubusercontent.com/assets/115071/10649474/38e9feea-77f9-11e5-9a6a-a47ba23b364b.png)



Themes:
![e-themes](https://cloud.githubusercontent.com/assets/115071/10649596/bcb35474-77f9-11e5-9a85-12773a384a55.png)

![themes](https://cloud.githubusercontent.com/assets/115071/10649475/3c49e6d6-77f9-11e5-8f81-a07096bdae86.png)



Menus
![e-menus](https://cloud.githubusercontent.com/assets/115071/10649591/b5d4c99e-77f9-11e5-9b50-e10ba14027cf.png)

![menus](https://cloud.githubusercontent.com/assets/115071/10649478/4131d5aa-77f9-11e5-841f-504785e02223.png)


Security Settings
![s-security](https://cloud.githubusercontent.com/assets/115071/10649593/b99c8814-77f9-11e5-87cd-6ff729dcce2e.png)

![security](https://cloud.githubusercontent.com/assets/115071/10649526/7dacf5e6-77f9-11e5-9e98-a4d9680f9790.png)


Default
![screen shot 2015-10-21 at 12 13 53 pm](https://cloud.githubusercontent.com/assets/115071/10649286/f50e1374-77f7-11e5-9ba1-aa44143639f6.png)

![default](https://cloud.githubusercontent.com/assets/115071/10649481/47bf6824-77f9-11e5-879d-2fdc384b0d24.png)

dereksmart commented:
This looks good to me -- Will merge after a designer signs off 

cc @alternatekev @justinkropp

justinkropp commented:
@dereksmart This looks good to merge, although we're going to be adding the requisite icons, correct?

dereksmart commented:
@justinkropp spoke with @enejb -- going to wait to merge until it has the icons

samhotchkiss commented:
LGTMT

enejb commented:
Added the icon. 
Updated the css to be part of jetpack-admin.scss

If there are no wording changes I think this is ready to merge. 

![screen shot 2015-10-26 at 4 13 20 pm](https://cloud.githubusercontent.com/assets/115071/10745861/85fda1ac-7c02-11e5-8a86-4f2c5336cd76.png)
![screen shot 2015-10-26 at 4 42 00 pm](https://cloud.githubusercontent.com/assets/115071/10745862/85fed2ca-7c02-11e5-9677-53c78d238c17.png)

lezama commented on modules/manage/activate-admin.php:
```diff
@@ -0,0 +1,95 @@
+	<p class="manage__description">
+		<?php esc_html_e( $description ); ?>
+	</p>
+	<p style="text-align: center;">	<?php printf( '<a class="button-primary manage__button" href="%1$s">%2$s</a>',
```
I would move the `a` tag outside of the php code

```
<a class="button-primary manage__button" href="<?php echo esc_url( $activate_url ) ?>" ><?php _e( 'Enable Jetpack Manage Now', 'jetpack'  ) ?></a>
```

lezama commented on modules/manage/confirm-admin.php:
```diff
@@ -0,0 +1,109 @@
+				'module' => 'manage',
+			)
+		),
+		"jetpack_activate-manage"
```
this should be single quotes

lezama commented on modules/manage/confirm-admin.php:
```diff
@@ -0,0 +1,109 @@
+		<?php esc_html_e( $description ); ?>
+	</p>
+	<p class="manage__description">
+		<?php printf( '<a class="manage__link"  href="%s">
```
Same thing here :)

lezama commented on modules/manage/confirm-admin.php:
```diff
@@ -0,0 +1,109 @@
+			</g>
+		</svg>
+	</div>
+	<h1 class="manage__title"><span class="genericon genericon-checkmark"></span><?php esc_html_e( __( 'Jetpack Manage Enabled', 'jetpack' ) ); ?></h1>
```
do we need to `esc_html` here?

enejb replied:
I think it is fine to leave it as it is.

lezama commented on modules/manage/activate-admin.php:
```diff
@@ -0,0 +1,95 @@
+				'module' => 'manage',
+			)
+		),
+		"jetpack_activate-manage"
```
single quotes!

lezama commented on class.jetpack.php:
```diff
@@ -4495,7 +4506,24 @@ public static function admin_screen_configure_module( $module_id ) {
 		// User that doesn't have 'jetpack_configure_modules' will never end up here since Jetpack Landing Page woun't let them.
 		if ( ! in_array( $module_id, Jetpack::get_active_modules() ) && current_user_can( 'manage_options' ) ) {
-			self::display_activate_module_link( $module_id );
+			if ( has_action( 'display_activate_module_setting_'.$module_id ) ) {
```
missing spaces around ` . `

lezama commented on class.jetpack.php:
```diff
@@ -4495,7 +4506,24 @@ public static function admin_screen_configure_module( $module_id ) {
+				 *
+				 * @param int $module_id Module ID.
+				 */
+				do_action( 'display_activate_module_setting_'.$module_id );
```
missing spaces around .

lezama commented on modules/manage/activate-admin.php:
```diff
@@ -0,0 +1,95 @@
+		<?php esc_html_e( $description ); ?>
+	</p>
+	<p class="manage__description">
+		<a class="button-primary manage__button" href="<?php echo esc_url( $activate_url ) ?>">
```
we could use `esc_html_e`

lezama commented on modules/manage/confirm-admin.php:
```diff
@@ -0,0 +1,109 @@
+		<?php esc_html_e( $description ); ?>
+	</p>
+	<p class="manage__description">
+		<a class="manage__link"  href="<?php echo esc_url( $link ); ?>">
```
`esc_html_e` FTW

lezama commented on modules/manage/confirm-admin.php:
```diff
@@ -0,0 +1,109 @@
+	</p>
+	<p class="manage__description">
+		<a class="manage__link"  href="<?php echo esc_url( $link ); ?>">
+			<span class="genericon genericon-previous"><?php echo esc_html( $link_title ); ?></span>
```
`esc_html_e` FTW

dereksmart commented:
Looks great with the icon

alternatekev commented:
There's a typo:

"Just one more step before *your* can manage..." it should be "before you can manage"

enejb commented:
Good catch @alternatekev. Fixed the typo in https://github.com/Automattic/jetpack/pull/2902

jeherve commented:
Just tested this, it seems to work but I'd have 2 remarks:

![screen shot 2015-10-28 at 2 13 29 pm](https://cloud.githubusercontent.com/assets/426388/10789504/164fdb72-7d7e-11e5-8e5f-2d8b9ce5a10c.png)

- [] The left arrow is a bit confusing imo. It implies that I'm going to "go back", while in fact it leads me to a different site altogether. How about a right arrow after the text?
- [] This text leads to `'https://wordpress.com/stats/day/' . $normalized_site_url;`, as per #2902. However, there are cases when the Stats module is disabled on the site. We should probably take this into account and redirect somewhere on WordPress.com.
